File: services/CandidateService/DTOs/UpdateCandidateProfileRequest.cs
using System.ComponentModel.DataAnnotations;
using CandidateService.Enums;

namespace CandidateService.DTOs;
public class UpdateCandidateProfileRequest
{
    [Required, StringLength(2000, MinimumLength = 20)]
    public string Bio { get; set; } = string.Empty;

    [Required, StringLength(120, MinimumLength = 2)]
    public string Location { get; set; } = string.Empty;
    public ExperienceLevel ExperienceLevel { get; set; }
    [MinLength(1)]
    public List<Skill> Skills { get; set; } = new();
    [MinLength(1)]
    public List<JobCategory> DesiredJobCategories { get; set; } = new();
}

File: frontend/src/types/candidate/UpdateCandidateProfileRequest.ts
import type { ExperienceLevel } from '../../models/jobs/ExperienceLevel';
import type { JobCategory } from '../../models/jobs/JobCategory';
import type { Skill } from '../../models/jobs/Skill';

export interface UpdateCandidateProfileRequest {
  bio: string;
  location: string;
  experienceLevel: ExperienceLevel;
  skills: Skill[];
  desiredJobCategories: JobCategory[];
}